### PR TITLE
Crypto Unit Tests

### PR DESCRIPTION
**Pull request by danyalKhaliq10p**
Unit tests created for Crypto library coverage using GCS
Signed-off-by: Naveed Jamil <naveed.jamil@tenpearls.com>

### CODE REVIEW

File: pkg/random/random_test.go [hunk 1]
```diff
@@ -20,3 +22,13 @@ func TestConcurrency(t *testing.T) {
...
 	wg.Wait()
 }
+
+func TestReader_Read(t *testing.T) {
+	var Reader io.Reader = &reader{rnd: Rand}
+	var s []byte = make([]byte, 5, 5)
+	n,err := Reader.Read(s)
+
+	require.Equal(t,n,5)
+	require.Nil(t,err)
```
Comment (uzairalikhan) [inline pkg/random/random_test.go]: use `require.NoError(t, err)` instead of Nil.

File: pkg/random/random_test.go [hunk 2]
```diff
@@ -20,3 +22,13 @@ func TestConcurrency(t *testing.T) {
...
+}
```
Comment (uzairalikhan) [inline pkg/random/random_test.go]: Also do we really need this test?

File: pkg/tlsconfig/tlsconfig_clone_test.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	"crypto/tls"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestClone(t *testing.T) {
+	cfg := tls.Config{}
+	config := Clone(&cfg)
+
+	assert.ObjectsAreEqualValues(cfg,*config)
+
```
Comment (uzairalikhan) [inline pkg/tlsconfig/tlsconfig_clone_test.go]: I don't think we really need this test, as just empty config is cloned and then checked for equality.

File: client/transport_test.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+		TLSClientConfig: &tls.Config{ServerName: "localhost"},
+	}
+	//defer tr.CloseIdleConnections()
+	client := http.Client{
+		Timeout: timeout,
+		Transport: tr,
+	}
+	tlsConfig := resolveTLSConfig(client.Transport)
+
+	require.NotNil(t,tlsConfig)
```
Comment (danyalKhaliq10p) [inline client/transport_test.go]: yes , on the next line the initially sent server name is checked for equality

File: pkg/random/random_test.go (hunk 2 above)
Comment (danyalKhaliq10p) [inline pkg/random/random_test.go]: this function was using crypto , thats why we required to have a test on this .

File: pkg/tlsconfig/tlsconfig_clone_test.go (hunk 3 above)
Comment (danyalKhaliq10p) [inline pkg/tlsconfig/tlsconfig_clone_test.go]: that has been updated , values are passed instead of empty tlsConfig

File: client/transport_test.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package client
+
+import (
+	"testing"
+	"net/http"
+	"time"
+	"github.com/stretchr/testify/require"
+	"crypto/tls"
```
Comment (danyalKhaliq10p) [inline client/transport_test.go]: done

File: client/transport_test.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+)
```
Comment (danyalKhaliq10p) [inline client/transport_test.go]: Done

File: client/transport_test.go [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+
+func TestResolveTLSConfig(t *testing.T){
```
Comment (danyalKhaliq10p) [inline client/transport_test.go]: Done
